test/functional/test_framework/script.py

```
@@ -662,9 +661,41 @@ def SignatureHash(script, txTo, inIdx, hashtype):
+    return s
+
+def SignatureHash(script, txTo, inIdx, hashtype):
+    """Consensus-correct SignatureHash
+
+    Returns (hash, err) to precisely match the consensus-critical behavior of
+    the SIGHASH_SINGLE bug. (inIdx is *not* checked for validity)
+    """
+    msg = LegacySignatureMsg(script, txTo, inIdx, hashtype)
+    return (hash256(msg), None)
+
+def LegacySignatureHash(*args, **kwargs):
```

Ruff (0.12.2)

Missing return type annotation for public function `LegacySignatureHash`

(ANN201)